### PR TITLE
4098 Users creating pages in joplin have departments

### PR DESCRIPTION
**Pull request by chiaberry**
# Description

Adds some tests around users creating pages. Tests that editors have department pages automatically assigned to pages they make and that Admins can create pages without departments. 

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 

# Checklist:
- [ ] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/pages/information_page/models.py [hunk 1]
```diff
@@ -1,16 +1,11 @@
 from django.db import models
 
-from modelcluster.fields import ParentalKey
-from modelcluster.models import ClusterableModel
```
Comment (chiaberry) [inline joplin/pages/information_page/models.py:4]: just clean up

File: joplin/users/tests/tests.py [hunk 2]
```diff
@@ -133,3 +137,63 @@ def test_make_superuser_with_roles_and_department(department):
...
+    HomePageFactory.create(**page_data)
+    # editor creates page
+    body = {
+            "type": "information",
+            "jwtToken": "",
+            "title": "Test Page",
+            "topic": "",
+            "department": "",
+    }
+    request = rf.post('admin/pages/new_from_modal', body, content_type='application/json')
```
Comment (chiaberry) [inline joplin/users/tests/tests.py:159]: rf is the RequestFactory

File: joplin/users/tests/tests.py [hunk 3]
```diff
@@ -133,3 +137,63 @@ def test_make_superuser_with_roles_and_department(department):
...
+
+@pytest.mark.django_db
+def test_admin_can_make_departmentless_page(superadmin, rf):
+    page_data = {
+        "publish_janis_branch_for_pr": "pytest",
+        "preview_janis_branch_for_pr": "pytest",
+        "slug": "pytest",
+        "title": "pytest",
+    }
+    HomePageFactory.create(**page_data)
```
Comment (chiaberry) [inline joplin/users/tests/tests.py:181]: should I make this each time? Or should I set up the homepage only once before running the tests? (if so, where do I put it?)

File: joplin/users/fixtures/helpers/create_fixture.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 
 # Skips creating fixture if User with email already exists
-def create_fixture(user_data, fixture_name):
+def create_fixture(user_data, fixture_name, group_list=[]):
```
Comment (briaguya) [inline joplin/users/fixtures/helpers/create_fixture.py:6]: Diggin the reusability!

File: joplin/users/tests/tests.py (hunk 3 above)
Comment (briaguya) [reply joplin/users/tests/tests.py:181]: For now it’s just the two times so I’d say it’s fine

File: joplin/users/fixtures/administrative/editor_for_test_env.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        "email": "editor@austintexas.io",
+        "is_superuser": False,
+        "password": os.getenv("API_TEST_USER_PASSWORD"),
+        "first_name": "Editor",
+        "last_name": "Robot",
+        "is_active": True,
+        "is_staff": True,
+    }
+
+    editor_group = Group.objects.get(name="Editors")
```
Comment (Niicck) [inline joplin/users/fixtures/administrative/editor_for_test_env.py:22]: Oh this is great. This is much more explicit than the hardcoded id=1 that I'd been using in other places.

File: joplin/users/tests/tests.py (hunk 2 above)
Comment (Niicck) [reply joplin/users/tests/tests.py:159]: That is an awesome discovery!

File: joplin/users/tests/tests.py (hunk 3 above)
Comment (Niicck) [reply joplin/users/tests/tests.py:181]: We've got a fixture for a dummy HomePage in pages/home_page/fixtures/test_cases/pytest.py

File: joplin/users/tests/tests.py [hunk 6]
```diff
@@ -133,3 +137,63 @@ def test_make_superuser_with_roles_and_department(department):
...
+            "topic": "",
+            "department": "",
+    }
+    request = rf.post('admin/pages/new_from_modal', body, content_type='application/json')
+    request.user = editor
+
+    # assert page is created
+    response = new_page_from_modal(request)
+    assert response.status_code is 200
+    # assert page has a department that matches editor department
```
Comment (Niicck) [inline joplin/users/tests/tests.py:165]: This is amazing! We had no way to test this before. This is great.

File: joplin/users/tests/tests.py (hunk 3 above)
Comment (chiaberry) [reply joplin/users/tests/tests.py:181]: I saw the fixture but without me creating this page, I'd get an error saying there was no home page

**Pull request merged**